src/wallet/wallet2.cpp

```diff
@@ -1351,12 +1356,15 @@ std::unique_ptr<wallet2> wallet2::make_dummy(const boost::program_options::varia
 }
 
 //----------------------------------------------------------------------------------------------------
-bool wallet2::set_daemon(std::string daemon_address, boost::optional<epee::net_utils::http::login> daemon_login, bool trusted_daemon, epee::net_utils::ssl_options_t ssl_options)
+bool wallet2::set_daemon(std::string daemon_address, boost::optional<epee::net_utils::http::login> daemon_login, bool trusted_daemon, epee::net_utils::ssl_options_t ssl_options, const std::string& proxy)
 {
   boost::lock_guard<boost::recursive_mutex> lock(m_daemon_rpc_mutex);
 
   if(m_http_client->is_connected())
     m_http_client->disconnect();
+  CHECK_AND_ASSERT_MES2(m_proxy.empty() || proxy.empty() , "It is not possible to set global proxy (--proxy) and daemon specific proxy together.");
+  if(m_proxy.empty())
```

Comment:
Shouldn't this be `if (!proxy.empty())`?

Reply:
I guess it depends if we want to behavior to be to reset the proxy when proxy is an empty string or not. I actually change my mind and like this behavior better. Let's keep it so that it resets every time.

Reply:
By the way. Keep in mind the global proxy is not changeable. That is kind of security requirement in a sense when user global proxy via --proxy flag. We should not be able to touch it, even if user asks. 


But the daemon specific proxy is daemon specific. Meaning you set proxy via RPC, to only that daemon. Later if you change daemon you have to set proxy to that daemon too.